### PR TITLE
Added support for `tasks` under `scheduling` .

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Added support for `tasks` under `scheduling` .

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-18112

### CODE REVIEW

File: service/ocean/providers/azure_np/common.go [hunk 1]
```diff
@@ -318,6 +319,35 @@ type ShutdownHours struct {
 	nullFields      []string
 }
 
+type Task struct {
+	IsEnabled      *bool      `json:"isEnabled,omitempty"`
+	Type           *string    `json:"taskType,omitempty"`
```
Comment (chandra1-n) [inline service/ocean/providers/azure_np/common.go:324]: modify to TaskType

File: service/ocean/providers/azure_np/common.go [hunk 2]
```diff
@@ -318,6 +319,35 @@ type ShutdownHours struct {
...
+	CronExpression *string    `json:"cronExpression,omitempty"`
+	Parameter      *Parameter `json:"parameters,omitempty"`
```
Comment (chandra1-n) [inline service/ocean/providers/azure_np/common.go:326]: modify to Parameters

File: service/ocean/providers/azure_np/common.go [hunk 3]
```diff
@@ -512,3 +542,104 @@ func (o *Filters) SetGpuTypes(v []string) *Filters {
...
+
+// region Parameter
+
+func (o *Task) SetParameter(v *Parameter) *Task {
+	if o.Parameter = v; o.Parameter == nil {
+		o.nullFields = append(o.nullFields, "Parameter")
+	}
+	return o
+}
+
```
Comment (chandra1-n) [inline service/ocean/providers/azure_np/common.go:592]: MarshalJSON() is missing for paramters object and clusterRoll object

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: service/ocean/providers/azure_np/common.go [hunk 4]
```diff
@@ -318,6 +319,35 @@ type ShutdownHours struct {
 	nullFields      []string
 }
 
+type Task struct {
```
[the remark this answers is not in the document]
Comment (sharadkesarwani) [reply service/ocean/providers/azure_np/common.go:322]: Updated

File: service/ocean/providers/azure_np/common.go (hunk 3 above)
Comment (sharadkesarwani) [reply service/ocean/providers/azure_np/common.go:592]: Added MarshalJSON() for both parameters and clusterRoll

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**